speculos/client.py

```diff
@@ -33,14 +33,32 @@ def check_status_code(response: requests.Response, url: str) -> None:
 
 class ApduResponse:
     def __init__(self, response: requests.Response) -> None:
-        self.response = response
-
-    def receive(self) -> bytes:
-        check_status_code(self.response, "/apdu")
-        data, status = split_apdu(bytes.fromhex(self.response.json()["data"]))
-        if status != 0x9000:
-            raise ApduException(status, data)
-        return data
+        self.__response = response
+
+    def __receive(self):
+        """ Get the response if not yet received and store it in `__data` and `__status` attributes """
+        if self.__response is not None:
+            check_status_code(self.__response, "/apdu")
+            self.__data, self.__status = split_apdu(bytes.fromhex(self.__response.json()["data"]))
+            self.__response = None
+
+    def data(self, expected_sw: int=0x9000) -> bytes:
+        """
+        :return: Received data (status word stripped)
+        :param expected_sw: Expected status word value.
+        :raises ApduException: If received status word is different from `expected_sw` parameter.
+        """
+        self.__receive()
+        if self.__status != 0x9000:
```

Comment:
This should be `if self.__status != expected_sw:`, if I understand correctly the aim of this Pull Request.

Reply:
Yes, fixed.